plugins/pulp_rpm/plugins/importers/yum/utils.py

```diff
@@ -99,3 +100,64 @@ def strip_ns(element, uri=None):
         element.tag = element.tag.replace('{%s}' % uri, '')
     for child in list(element):
         strip_ns(child, uri)
+
+
+class RepoURLModifier(object):
+    """
+    Repository URL Modifier
+
+    :ivar conf: URL modifier persistent configuration
+    :type conf: dict
+
+    """
+    def __init__(self, **kwargs):
+        """
+        :ivar conf: URL modifier persistent configuration, populated from optional keyword
+                    arguments (see :py:meth:`RepoURLModifier.__call__` for allowed options).
+        :type conf: dict
+
+        """
+        self.conf = kwargs
```

Comment:
I would prefer to be explicit about the allowed options here so that typos will raise a TypeError. Nectar used to work similarly to this and it made debugging typos a little more difficult.